src/common/IObject.hpp

```diff
@@ -53,6 +53,25 @@ class IObject {
  [[nodiscard]]
  virtual std::vector<IComponent *> getComponents() const = 0;
 
+ /**
+  * @brief Returns a component of the object
+  * @tparam T The type of the component to get
+  * @return A pointer to the component if it exists, nullptr otherwise
+  * @version v0.1.0
+  * @since v0.1.0
+  * @author Landry GIGANT
+  */
+ template<class T>
+ [[nodiscard]]
+ T *getComponent() const {
```

Comment:
This functions shall be useless since the only IObject will be VirtualObject, and it contains no logic. If you plan to use this function, you either have made a mistake or could have used AComponent::getParentComponent<>();